### PR TITLE
Lava api errors

### PR DESCRIPTION
**Pull request by mwasilew**
This is a series of patches that deals with errors resulting from using LAVA's REST API.

### CODE REVIEW

File: test/ci/backend/test_lava.py [hunk 1]
```diff
@@ -1257,3 +1258,21 @@ def test_resubmit_job_lava_error(self, __resubmit__):
...
+        testjob = TestJob(
+            definition=test_definition,
+            backend=self.backend,
+            target=self.project,
+            submitted=True,
+            job_status='Canceled',
+            job_id="1236"
+        )
+        with requests_mock.Mocker() as m:
+            m.post("http://example.com/api/v0.2/jobs/1236/resubmit",
```
Comment (chaws) [inline test/ci/backend/test_lava.py:1275]: I saw this error: `requests_mock.exceptions.NoMockAddress: No mock address: POST http://example.com/api/v0.2/jobs/1236/resubmit/`

Maybe `requests_mock` is adding the extra slash there? And it seems the requested URL and the mocked one need to be exactly the same: https://stackoverflow.com/questions/62981932/requests-mock-nomockaddress-exception

File: squad/ci/backend/lava.py [hunk 2]
```diff
@@ -357,7 +357,7 @@ def __resubmit__(self, job_id):
         if self.use_xml_rpc:
             return self.proxy.scheduler.resubmit_job(job_id)
         response = requests.post(
-            urljoin(self.api_url_base, "jobs/%s/resubmit" % (job_id)),
+            urljoin(self.api_url_base, "jobs/%s/resubmit/" % (job_id)),
```
Comment (chaws) [inline squad/ci/backend/lava.py:360]: While you're at it, could you please also add the trailing slash here https://github.com/Linaro/squad/blob/b2c17c7c333b37a4a95d33c9ce75fbdc96c98a42/squad/ci/backend/lava.py#L377

File: test/ci/backend/test_lava.py (hunk 1 above)
Comment (mwasilew) [reply test/ci/backend/test_lava.py]: it's my mistake. I forgot to update the address when adding a trailing slash into the code.

File: squad/ci/backend/lava.py (hunk 2 above)
Comment (mwasilew) [reply squad/ci/backend/lava.py:360]: I guess I can. This one isn't affected by redirect as it's GET not POST.

File: test/ci/backend/test_lava.py [hunk 3]
```diff
@@ -1243,3 +1244,35 @@ def test_resubmit_canceled_job(self, __resubmit__):
...
+            definition=test_definition,
+            backend=self.backend,
+            target=self.project,
+            submitted=True,
+            job_status='Canceled',
+            job_id="1236"
+        )
+        with requests_mock.Mocker() as m:
+            m.post("http://example.com/api/v0.2/jobs/1236/resubmit/",
+                    status_code=405,
```
Comment (chaws) [inline test/ci/backend/test_lava.py:1276]: I think flake8 is complaining about the indentation here. My guess is that it's expecting 1 tab, but there's 2.

**Comment by chaws** [comment]
LGTM, thanks for the patch @mwasilew !

**Pull request merged**